pkg/route/api/validation/validation.go

```diff
@@ -135,3 +140,35 @@ func validateNoDoubleEscapes(tls *routeapi.TLSConfig) fielderrors.ValidationErro
 	}
 	return allErrs
 }
+
+// validateTLSInsecure tests fields for different types of insecure options.
+// Called by validateTLS.
+func validateTLSInsecure(tls *routeapi.TLSConfig) *fielderrors.ValidationError {
+	// Check insecure option value if specified (empty is ok).
+	if tls == nil || tls.Insecure == "" {
+		return nil
+	}
+
+	// Ensure insecure is set only for edge terminated routes.
+	if routeapi.TLSTerminationEdge != tls.Termination {
+		// tls.Insecure option is not supported for a non edge-terminated routes.
+		msg := fmt.Sprintf("invalid value %s for insecure option - insecure is only supported for edge-terminated routes", tls.Insecure)
+		return fielderrors.NewFieldInvalid("insecure", tls.Insecure, msg)
+	}
+
+	// It is an edge-terminated route, check insecure option value is
+	// one of Disable/Expose/Redirect.
+	switch tls.Insecure {
```

Comment:
put the supported values in a string set, check if it's in that set, then do NewFieldValueNotSupported with the allowed values